lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -259,6 +259,39 @@ def tools_on_multiple_hosts(self, pbench_run):
                     tool = p / "perf"
                     tool.write_text("--record-opts='-a --freq=100'")
 
+    # Issue #3454
+    @pytest.fixture
+    def labels_on_multiple_hosts(self, pbench_run, tools_on_multiple_hosts):
+        # This fixture is meant to be called after the previous one
+        # (tools_on_multiple_hosts). The previous one establishes a
+        # tool-like directory structure; this one just embellishes it
+        # with labels on some host entries. Think of it as a
+        # double for-loop, like the one above, only unrolled.
```

Comment:
I would rework this text, to something like this:
```python
        # This fixture embellishes the structure produced by
        # tools_on_multiple_hosts, which establishes a tool-like directory
        # structure, by adding labels on some host entries. Think of it as a
        # double for-loop, like the one in tools_on_multiple_hosts, only unrolled.
```
That is, this fixture implicitly invokes `tools_on_multiple_hosts` already, so we don't want to suggest that it needs to be called; and, words like "previous" and "above" won't tolerate code refactoring and rearrangement, so I recommend avoiding them.